### PR TITLE
Add Support for Pascal

### PR DESCRIPTION
**Pull request by Coldzer0**

### CODE REVIEW

File: gen.py [hunk 1]
```diff
@@ -210,9 +244,10 @@ def end_enum(self):
             *children, = [x.displayname for x in c.get_children()]
             skip_prefix = len(os.path.commonprefix(children))
 
-            for x in c.get_children():
+            max = sum(1 for _ in c.get_children())
+            for index, x in  enumerate(list(c.get_children())):
```
Comment (athre0z) [inline gen.py]: This code currently walks the children twice and then still pulls the full list to memory (`list(..)`). I think we should just collect the list once instead which then allows us to just `len(x)` it. Something like:

```suggestion
            enum_members = list(c.get_children())
            max = len(enum_members)
            for index, x in enumerate(enum_members):
```

File: gen.py [hunk 2]
```diff
@@ -210,9 +244,10 @@ def end_enum(self):
...
                 name = x.displayname[skip_prefix:]
                 if name[0].isdigit() or name in mode.reserved_keywords:
                     name = "_" + name
-                mode.enum_member(name, x.displayname, x.enum_value, x.brief_comment)
+                mode.enum_member(name, x.displayname, x.enum_value, x.brief_comment, (index == max - 1) )
```
Comment (athre0z) [inline gen.py]:
```suggestion
                mode.enum_member(name, x.displayname, x.enum_value, x.brief_comment, index == max - 1)
```

**Review by athre0z — COMMENTED** [review]
Looks good in general -- just two nits. Thanks for the PR!

File: gen.py (hunk 1 above)
Comment (Coldzer0) [reply gen.py]: Looks good; I'll commi these changes.

Sorry, I don't use Python that much

**Review by athre0z — APPROVED** [review]
LGTM!

**Pull request merged**